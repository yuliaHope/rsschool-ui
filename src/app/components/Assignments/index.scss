@media (min-width: 576px) {
    .card {
        min-width: 44%;
    }
}
@media (min-width: 1000px) {
    .card {
        min-width: 30%;
    }
}
.score{
    float: right;
    font-weight: bold;
}

.card-footer {
    background: none;
    border: none;
}

.studentComment{
    resize: none;
    height: 85px;
}

.card-deck::after{
    content: "";
    width: 30%;
    margin: 0 15px;
}