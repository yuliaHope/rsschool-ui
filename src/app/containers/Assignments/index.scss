.badge {
    padding: 3px 4px 1px;
    font-size: .875rem;
    line-height: 1.5;
}