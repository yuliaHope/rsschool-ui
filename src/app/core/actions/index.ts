export * from './schedule';
export * from './user';
export * from './courses';
export * from './assignments';
