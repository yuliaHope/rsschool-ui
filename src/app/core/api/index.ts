export * from './schedule';
export * from './session';
export * from './user';
export * from './courses';
export * from './assignments';
