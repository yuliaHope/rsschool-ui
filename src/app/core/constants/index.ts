export * from './schedule';
export * from './courses';
export * from './user';
export * from './assignments';
