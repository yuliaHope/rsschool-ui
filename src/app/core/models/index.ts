export * from './events';
export * from './stage';
export * from './user';
export * from './course';
export * from './courseMentor';
export * from './courseStudent';
export * from './assignment';
