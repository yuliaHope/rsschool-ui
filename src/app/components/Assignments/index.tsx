import * as React from 'react';
import AssignmentItem from './AssignmentItem/AssignmentItem';
import { INormalizeAssignment } from 'core/reducers/assignments';
import { classNames } from 'core/styles';

const cn = classNames(require('./index.scss'));

type AssignmentsProps = {
    courseId: string;
    assignments: INormalizeAssignment[];
    updateAssignment: (assignment: INormalizeAssignment) => void;
};

const Assignments = (props: AssignmentsProps) => {
    const { assignments } = props;
    return (
        <div className={`${cn('card-deck')} card-deck mb-3 justify-content-between`}>
            {assignments
                ? assignments.map((assignment: INormalizeAssignment) => {
                      if (assignment.isActiveAssignment) {
                          return (
                              <AssignmentItem
                                  key={assignment.assignment._id}
                                  assignment={assignment}
                                  updateAssignment={props.updateAssignment}
                              />
                          );
                      } else {
                          return null;
                      }
                  })
                : null}
        </div>
    );
};

export default Assignments;
