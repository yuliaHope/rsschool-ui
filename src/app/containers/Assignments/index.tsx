import * as React from 'react';
import { connect } from 'react-redux';
import { Row, Col, Badge } from 'reactstrap';
import Assignments from 'components/Assignments';
import { fetchAssignments, updateAssignment } from 'core/actions';
import { classNames } from 'core/styles';
import './index.scss';
import { INormalizeAssignment } from 'core/reducers/assignments';
import { RootState } from 'core/reducers';

const cn = classNames(require('./index.scss'));

const mapStateToProps = (state: RootState, props: any): AssignmentsContainerProps => {
    if (state.assignments == null || state.user == null) {
        return props;
    }
    return {
        ...props,
        courseId: props.match.params.id,
        isAdmin: state.user.isAdmin,
        assignments: state.assignments.assignments,
        isLoading: state.assignments.isLoading,
    };
};

const mapDispatchToProps = (dispatch: any, props: any): AssignmentsContainerProps => {
    return {
        ...props,
        onLoad: id => {
            dispatch(fetchAssignments(id));
        },
        updateAssignment: (assignment: INormalizeAssignment) => {
            dispatch(updateAssignment(assignment));
        },
    };
};

type AssignmentsContainerProps = {
    onLoad: (id: string) => void;
    courseId: string;
    isAdmin: boolean;
    userProfile: string;
    assignments: INormalizeAssignment[];
    isLoading: boolean;
    updateAssignment: (assignment: INormalizeAssignment) => void;
};

class AssignmentsContainer extends React.Component<AssignmentsContainerProps> {
    componentDidMount() {
        this.props.onLoad(this.props.courseId);
    }

    getFullScore(): number {
        const { assignments } = this.props;
        let score = 0;
        assignments.forEach(assignmentItem => {
            score += assignmentItem.assignment.score;
        });
        return score;
    }

    render() {
        const { courseId, assignments, isLoading, isAdmin } = this.props;
        return (
            !isAdmin && (
                <div className="tasks">
                    <h2>TASKS</h2>
                    <Row>
                        <Col xs="6">
                            <p>
                                Your github private repository
                                <Badge color="dark" href="#" className={cn('badge')}>
                                    {' '}
                                    here
                                </Badge>
                            </p>
                        </Col>
                        <Col xs="6" className="text-right">
                            <p>You are in the TOP 50 students!</p>
                            <p>Full Score: {this.getFullScore()}</p>
                        </Col>
                    </Row>
                    {isLoading ? (
                        <h3>Loading...</h3>
                    ) : (
                        <Assignments
                            courseId={courseId}
                            assignments={assignments}
                            updateAssignment={this.props.updateAssignment}
                        />
                    )}
                </div>
            )
        );
    }
}

export default connect(
    mapStateToProps,
    mapDispatchToProps,
)(AssignmentsContainer);
