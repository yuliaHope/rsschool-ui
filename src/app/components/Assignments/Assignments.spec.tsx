import * as React from 'react';
import { shallow } from 'enzyme';
import Assignments from './index';
import { AssignmentStatus } from '../../core/models';

describe('Assignments', () => {
    const props = {
        courseId: 'courseId',
        assignments: [],
        updateAssignment: jest.fn(),
    };

    describe('render with no data', () => {
        it('renders properly', () => {
            const output = shallow(<Assignments {...props} />);
            expect(output).toMatchSnapshot();
        });
    });

    describe('render with data', () => {
        const nextProps = {
            ...props,
            assignments: [
                {
                    isEndAssignment: false,
                    isActiveAssignment: true,
                    assignment: {
                        _id: 123,
                        assignmentRepo: 'assignmentRepo',
                        checkDate: 1555555555555555555555,
                        completeDate: 10000555555555555,
                        courseId: 'courseId',
                        deadlineDate: 15555555555555555555555,
                        mentorComment: 'mentorComment',
                        mentorId: 'mentorId',
                        score: 100,
                        status: AssignmentStatus.Assigned,
                        studentComment: 'studentComment',
                        studentId: 'studentId',
                        taskId: 'taskId',
                        title: 'title',
                        urlToDescription: 'urlToDescription',
                        whoChecks: 'whoChecks',
                    },
                },
            ],
        };
        it('renders properly', () => {
            const output = shallow(<Assignments {...nextProps} />);
            expect(output).toMatchSnapshot();
        });
    });
});
