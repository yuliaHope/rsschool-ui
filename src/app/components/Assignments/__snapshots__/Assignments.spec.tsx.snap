// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assignments render with data renders properly 1`] = `
<div
  className="card-deck card-deck mb-3 justify-content-between"
>
  <AssignmentItem
    assignment={
      Object {
        "assignment": Object {
          "_id": 123,
          "assignmentRepo": "assignmentRepo",
          "checkDate": 1.5555555555555557e+21,
          "completeDate": 10000555555555556,
          "courseId": "courseId",
          "deadlineDate": 1.5555555555555555e+22,
          "mentorComment": "mentorComment",
          "mentorId": "mentorId",
          "score": 100,
          "status": "Assigned",
          "studentComment": "studentComment",
          "studentId": "studentId",
          "taskId": "taskId",
          "title": "title",
          "urlToDescription": "urlToDescription",
          "whoChecks": "whoChecks",
        },
        "isActiveAssignment": true,
        "isEndAssignment": false,
      }
    }
    key="123"
    updateAssignment={[MockFunction]}
  />
</div>
`;

exports[`Assignments render with no data renders properly 1`] = `
<div
  className="card-deck card-deck mb-3 justify-content-between"
/>
`;
